test/ColorLineTracingTest.cpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * @file   ColorLineTracingTest.cpp
+ * @brief  ColorLineTracingクラスのテスト
+ * @author YKhm20020
+ */
+
+#include "LineTracing.h"
+#include "ColorLineTracing.h"
+#include "Mileage.h"
+#include <gtest/gtest.h>
+#include <gtest/internal/gtest-port.h>
+
+using namespace std;
+
+namespace etrobocon2023_test {
+  TEST(GTestInit2, init)
+  {
+    const ePortS colorSensorPort = PORT_2;
+    const ePortS sonarSensorPort = PORT_3;
+    const ePortM armMotorPort = PORT_A;
+    const ePortM rightMotorPort = PORT_B;
+    const ePortM leftMotorPort = PORT_C;
+
+    ev3api::ColorSensor _colorSensor(colorSensorPort);
+    ev3api::SonarSensor _sonarSensor(sonarSensorPort);
+    ev3api::Motor _rightMotor(rightMotorPort);
+    ev3api::Motor _leftMotor(leftMotorPort);
+    ev3api::Motor _armMotor(armMotorPort);
+
+    Controller::rightMotor = &_rightMotor;
+    Controller::leftMotor = &_leftMotor;
+    Controller::armMotor = &_armMotor;
+    Measurer::colorSensor = &_colorSensor;
+    Measurer::sonarSensor = &_sonarSensor;
+    Measurer::rightMotor = &_rightMotor;
+    Measurer::leftMotor = &_leftMotor;
+    Measurer::armMotor = &_armMotor;
+
+    SUCCEED();
+  }
+
+  // 最初3回の色取得で連続して指定色を取得するテストケース
+  TEST(ColorLineTracingTest, runToGetFirst)
+  {
+    COLOR targetColor = COLOR::GREEN;
+    double targetSpeed = 50.0;
+    double targetBrightness = 50.0;
+    int basePwm = 100;
+    PidGain gain = { 0.1, 0.05, 0.05 };
+    bool isLeftEdge = true;
+    ColorLineTracing cl(targetColor, targetSpeed, targetBrightness, gain, isLeftEdge);
+
```

Comment:
モータカウントを比較するようなテストの場合は、以下の処理を追加すれば、モータカウントが安定するはずです。
事前条件を満たさない場合のテストでは省略可能。

```suggestion
Mesuarer::leftMotor->reset();
Mesuarer::rightMotor->reset();
```

Mesuarerで失敗したテストに関しても似たような変更で解決できると思いますが、Straight系統で変更されてるので、コンフリクトが面倒なので、Starightがマージされたmainを取り込む形がいいと思います。

Reply:
ありがとうございます、すべてのテストをパスすることができました🙇
基本動作の確認に入れそうです👍